crates/e2e/tests/e2e/quoting.rs

```diff
@@ -223,3 +241,144 @@ async fn uses_stale_liquidity(web3: Web3) {
+            sell_amount: SellAmount::BeforeFee {
+                value: NonZeroU256::try_from(to_wei(1)).unwrap(),
+            },
+        },
+        timeout,
+        ..Default::default()
+    };
+
+    /// The default and maximum quote timeout enforced by the backend.
+    /// (configurable but always 500ms in e2e tests)
+    const MAX_QUOTE_TIME_MS: u64 = 500;
+
+    let assert_within_variance = |start_timestamp: Instant, target| {
+        const VARIANCE: u64 = 100; // small buffer to allow for variance in the test
+        const HTTP_BUFFER: u64 = 10;
+        let min = target - HTTP_BUFFER;
+        let max = min + VARIANCE;
+        let elapsed = start_timestamp.elapsed().as_millis() as u64;
+        tracing::debug!(target, actual = ?elapsed, "finished request");
+        assert!((min..max).contains(&elapsed));
+    };
+
+    // native token price requests are also capped to the max timeout
+    let start = std::time::Instant::now();
+    let _ = services.get_native_price(&sell_token.address()).await;
+    assert_within_variance(start, MAX_QUOTE_TIME_MS);
+
+    // not providing a timeout uses the backend's default timeout (500ms)
+    let start = std::time::Instant::now();
+    let _ = services.submit_quote(&quote_request(None)).await;
+    assert_within_variance(start, MAX_QUOTE_TIME_MS);
+
+    // timeouts below the max timeout get enforced correctly
+    let start = std::time::Instant::now();
+    let _ = services
+        .submit_quote(&quote_request(Some(Duration::from_millis(300))))
+        .await;
+    assert_within_variance(start, 300);
+
+    // user provided timeouts get capped at the backend's max timeout (500ms)
+    let start = std::time::Instant::now();
+    let _ = services
+        .submit_quote(&quote_request(Some(Duration::from_millis(10_000))))
+        .await;
+    assert_within_variance(start, MAX_QUOTE_TIME_MS);
+
+    // set up trader to pass balance checks during order creation
+    sell_token.mint(trader.address(), to_wei(1)).await;
+    tx!(
+        trader.account(),
+        sell_token.approve(onchain.contracts().allowance, to_wei(1))
+    );
+
+    let order = OrderCreation {
+        sell_token: sell_token.address(),
+        sell_amount: to_wei(1),
+        buy_token: Default::default(),
+        buy_amount: to_wei(1),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        partially_fillable: true,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+
+    // order creation requests always use the default quote time
+    // to maximize the chance for the request to succeed because
+    // we return an error if we can't get a quote
+    let start = std::time::Instant::now();
+    let _ = services.create_order(&order).await;
+    assert_within_variance(start, MAX_QUOTE_TIME_MS);
+}
```

Comment:
We could also test with a very small timeout and assert the error code/message we get when no quotes are available in time